app/Resources/views/full/blog.html.twig

```diff
@@ -37,15 +38,15 @@
                                     </div>
                                 </div>
                             </div>
-                            <hr class="space-top" />
+                            <hr class="space-top">
                         </div>
                     </div>
                 </div>
 
                 <div class="container"><div class="row"><div class="col-md-10 col-md-offset-1 space-bottom"><div class="row">
             {% else %}
                 <div class="{% if loop.index % 2 == 1 %}col-md-offset-2 {% endif %}col-md-5 space-top">
-                    {{ render_esi(controller('ez_content:viewLocation', {
+                    {{ render_esi(controller('ez_content:viewContent', {
                         locationId: post.contentInfo.mainLocationId,
```

Comment:
Interesting question.

The issue with only using `contentId` is that any content item might have several locations. If you don't specify the `locationId`, the main one will be used, and it is probably not what you want.

The only problem with `locationId` alone is that it will be used to determine the `contentId`, which might be an issue in case of locationless content. But in the context of a search like you're using now, it shouldn't be a problem.

But on top of that, after checking the code again, _both_ `locationView` and `contentView` are deprecated in favour of `viewAction`. Can you replace those calls with `viewAction` ?